openshift/default-catalog-consistency/pkg/check/image.go

```diff
@@ -0,0 +1,111 @@
+package check
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/containers/image/v5/manifest"
+
+	specsgov1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+)
+
+// AllImageChecks returns a list of image checks to be performed on the image.
+func AllImageChecks() []ImageCheck {
+	return []ImageCheck{
+		ImageIsValidManifest(),
+		ImageHasExpectedLabelsWith(map[string]string{
+			"operators.operatorframework.io.index.configs.v1": "/configs",
+		}),
+	}
+}
+
+// ImageIsValidManifest checks if the image is a valid manifest.
+func ImageIsValidManifest() ImageCheck {
+	return ImageCheck{
+		Name: "ImageIsValidManifest",
+		Fn: func(ctx context.Context, root specsgov1.Descriptor, target oras.ReadOnlyTarget) error {
+			manifestReader, err := target.Fetch(ctx, root)
+			if err != nil {
+				return err
+			}
+			defer manifestReader.Close()
+			var imgManifest specsgov1.Manifest
+			if err := json.NewDecoder(manifestReader).Decode(&imgManifest); err != nil {
+				return err
+			}
+			switch imgManifest.MediaType {
+			case specsgov1.MediaTypeImageManifest, manifest.DockerV2Schema2MediaType:
+			default:
+				return fmt.Errorf("unrecognized manifest type: %s", imgManifest.MediaType)
+			}
+			return nil
+		},
+	}
+}
+
+// ImageHasExpectedLabelsWith checks if the image has the expected labels.
+func ImageHasExpectedLabelsWith(expectedLabels map[string]string) ImageCheck {
```

Comment:
Maybe just `ImageHasLabels`?

Reply:
Changed for `ImageHasLabelsWith` 
Just keep the With to make clear that it has args.

Reply:
👍 done